File: vol132/13276.py
# TLE
from math import ceil

def ans( E, P, K, R ):
	if P * K >= E:
		return ceil( E / P )
	elif P * K <= R:
		return -1
	else:
		round = ceil( ( E - P * K ) / ( P * K - R ) )
		return round * K + ceil( ( E - ( P * K - R ) * round ) / P )

for x in range( int( input() ) ):
	E, P, K, R = map( int, input().split() )
	print( 'Case %d: %d' % ( x + 1, ans( E, P, K, R ) ) )

File: vol132/13276.cpp
#include<cstdio>
#include<cmath>

double ans( double e, double p, double k, double r )
{
	if( p * k >= e )
		return ceil( e / p );
	else if( p * k <= r )
		return -1;
	double round = ceil( ( e - p * k ) / ( p * k - r ) );
	return round * k + ceil( ( e - ( p * k - r ) * round ) / p );
}

int main()
{
	int n, e, p, k, r;

	scanf( "%d", &n );
	for( int i = 1; i <= n && scanf( "%d %d %d %d", &e, &p, &k, &r ); ++i )
		printf( "Case %d: %.0f\n", i, ans( e, p, k, r ) );
}
